Can't compile with --debugger:on and use asyncdispatch at the same time?

When compiling this program:
```nim
import asyncdispatch, asyncnet

discard newAsyncSocket().accept
```
I get this error while it's being compiled:

> Error: execution of an external compiler program 'gcc -c  -w  -I/home/dawkot/.choosenim/toolchains/nim-0.19.2/lib -o /home/dawkot/.cache/nim/hello_d/stdlib_asyncdispatch.c.o /home/dawkot/.cache/nim/hello_d/stdlib_asyncdispatch.c' failed with exit code: 1
> 
> /home/dawkot/.cache/nim/hello_d/stdlib_asyncdispatch.c: In function ‘cb_vGUeE9b2tnecMPnZ7KQqTcA’:
> /home/dawkot/.cache/nim/hello_d/stdlib_asyncdispatch.c:1112:3: error: ‘else’ without a previous ‘if’
>    else {
>    ^

Even if debugger is not supported for asynchronous code a proper error would be nice.

--- a/compiler/ccgstmts.nim
+++ b/compiler/ccgstmts.nim
@@ -124,13 +124,14 @@ proc endBlock(p: BProc, blockEnd: Rope) =
 
 proc endBlock(p: BProc) =
   let topBlock = p.blocks.len - 1
-  var blockEnd = if p.blocks[topBlock].label != nil:
-      ropecg(p.module, "} $1: ;$n", p.blocks[topBlock].label)
-    else:
-      ~"}$n"
   let frameLen = p.blocks[topBlock].frameLen
+  var blockEnd: Rope
   if frameLen > 0:
     blockEnd.addf("FR_.len-=$1;$n", [frameLen.rope])
+  if p.blocks[topBlock].label != nil:
+    blockEnd.addf("} $1: ;$n", [p.blocks[topBlock].label])
+  else:
+    blockEnd.addf("}$n", [])
   endBlock(p, blockEnd)
 
 proc genSimpleBlock(p: BProc, stmts: PNode) {.inline.} =

